pkgs/servers/flaresolverr/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib, stdenv, python3, chromium, xvfb-run, xorgserver, makeWrapper, chromedriver, fetchFromGitHub }:
+
+let
+  python_env = python3.withPackages
+    (p: with p; [ bottle waitress selenium func-timeout requests websockets xvfbwrapper webtest ]);
+in
+stdenv.mkDerivation rec {
+  pname = "flaresolverr";
+  version = "3.0.2";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-zpeJf1CaQ4bsncZz44sH+tFKddYrZf7YdNYL50d9GA4=";
+  };
+
+  buildInputs = [ makeWrapper ];
+  nativeBuildInputs = [ chromedriver ];
+  patches = [ ./chromedriver_path.patch ];
+
+  postPatch = ''
+    substituteInPlace src/utils.py \
+    --replace "NIX_STORE_DRIVER_PATH" "${chromedriver}/bin/chromedriver"
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share
+    cp -r src $out/share
+    cp package.json $out/share
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    makeWrapper ${python_env}/bin/python $out/bin/flaresolverr \
+      --prefix PATH : ${lib.makeBinPath [ chromium xvfb-run xorgserver chromedriver ]} \
+      --add-flags $out/share/src/flaresolverr.py \
+      --chdir $out/share/
+  '';
+
+  meta = with lib; {
+    description = "Proxy server to bypass Cloudflare protection";
+    homepage = "https://github.com/FlareSolverr/FlareSolverr";
+    license = licenses.mit;
+    maintainers = with maintainers; [ julienmalka ];
+    platforms = [ "x86_64-linux" ];
```

Comment:
this seems to support at least `*-linux` for `*` reasonable (i.e. i686, x86_64, armv7, aarch64)?

Reply:
Nope, exact support is inter(chromedriver, xvfb-run) which is exactly x86_64-linux. I can write that in the plateforms field is you think it is useful.

Reply:
Hm, IMHO, you should not encode dependency platforms in the final program.
You should encode the intrinsic platform support for this package, because otherwise, you will cripple it artificially.
If a dependency (chromedriver) is unsupported, it will break at this step rather than here.
You also enable a user to provide a stand-in replacement.

Reply:
Well flaresolverr is only a Python project, so its intrinsic platform support is ``platforms.all``. I am not sure it is helpful for users though because they will expect that if their platform is marked as supported then they can run the software, no?

Reply:
I guess we can settle for `lib.intersectLists chromedriver.platforms xvfb-run.platforms` and add a comment mentioning this situation I guess.

Reply:
Working for me.